components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1358,13 +1358,9 @@ private long getAuthenticatedTimeFromCommonAuthCookie(Cookie cookie) {
             String sessionContextKey = DigestUtils.sha256Hex(cookie.getValue());
             SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(sessionContextKey);
             if (sessionContext != null) {
-                if (sessionContext.getProperty(FrameworkConstants.UPDATED_TIMESTAMP) != null) {
-                    authTime = Long.parseLong(
-                            sessionContext.getProperty(FrameworkConstants.UPDATED_TIMESTAMP).toString());
-                } else {
-                    authTime = Long.parseLong(
+                // Auth_time should be the first time user authenticated giving credentials.
+                authTime = Long.parseLong(
```

Comment:
Why to convert a Long to String and then Parse back to String?
It was the way done earlier. But can be improved now.